server/src/storage/segment.rs

```diff
@@ -41,6 +41,8 @@ pub trait UncompressedSegment {
     fn get_record_batch(&mut self) -> RecordBatch;
 
     fn get_memory_size(&self) -> usize;
+
+    fn spill_to_parquet(&mut self, key: String) -> Result<SpilledSegment, String>;
```

Comment:
Maybe add a comment documenting why `spill_to_parquet` is defined on `UncompressedSegment` instead of `SegmentBuilder`. Also, maybe return [`std::io::Error`](https://doc.rust-lang.org/std/io/struct.Error.html) instead of `String`.

Reply:
Fixed.